api/functions/utils/clean_json.py

```diff
@@ -0,0 +1,206 @@
+    else:
+        english_words = set()
+
+    # Clean the text in the dict, returns a list of cleaned words
+    clean_words, english_word_count = clean_utterance(
+        utterance=utterance,
+        punctuation_to_collapse_by=punctuation_to_collapse_by,
+        punctuation_to_explode_by=punctuation_to_explode_by,
+        special_cases=special_cases,
+        translation_tags=translation_tags,
+        remove_english=remove_english,
+        english_words=english_words,
+    )
+
+    # Check that the cleaned words are valid (ie, not null, not English etc)
+    if are_words_valid(clean_words, english_word_count, remove_english):
+        cleaned_transcript = " ".join(clean_words).strip()
+    else:
+        # TODO is it best to return an empty str here or raise an error?
+        cleaned_transcript = ""
+
+    utterance["transcript"] = cleaned_transcript
+    return utterance
+
+
+def clean_utterance(
+    utterance: Dict[str, str],
+    punctuation_to_collapse_by: str = "",
+    punctuation_to_explode_by: str = "",
+    special_cases: set = {None},
+    translation_tags: set = {None},
+    remove_english: bool = False,
+    english_words: set = None,
+) -> Tuple[List[str], int]:
+    """
+    Takes an utterance and cleans it based on the rules established by the provided parameters.
+    :param utterance: a dictionary with a "transcript" key-value pair.
+    :param punctuation_to_collapse_by: punctuation marks to strip.
+    :param punctuation_to_explode_by: punctuation marks to replace with spaces.
+    :param special_cases: a list of words to always remove from the output.
+    :param translation_tags: a list of tags to always remove from the output.
+    :param remove_english: whether or not to remove English words.
+    :param english_words: a list of english words to remove from the transcript (we suggest the nltk corpus).
+    :return: a tuple with a list of 'cleaned' words and the number of English to remove.
+    """
+    # TODO add interface setting to include user specific tags
+    # translation_tags = {"@eng@", "<ind:", "<eng:"}
+    # TODO add interface setting to skip this as caps are significant in some languages
+    utterance_string = utterance.get("transcript").lower()
+    dirty_words = utterance_string.split()
+    clean_words = []
+    english_word_count = 0
+    for word in dirty_words:
+        if word in special_cases:
+            continue
+        if remove_english and len(word) > 3 and word in english_words:
+            english_word_count += 1
+            continue
+        if word in translation_tags:
+            return [], 0
+        # Word is ok to use, now clean it
+        word = deal_with_punctuation(
+            text=word,
+            punctuation_to_collapse_by=punctuation_to_collapse_by,
+            punctuation_to_explode_by=punctuation_to_explode_by,
+        )
+        clean_words.append(word)
+    return clean_words, english_word_count
+
+
+def get_english_words() -> Set[str]:
+    """ """
```

Comment:
lol wat 😂

Reply:
Oh no, I've been caught!
- Previously this was a call which would download a set of english words from an online source.
- We weren't currently using this feature in the processing options, and I didn't want to have to add an unused dependency at this stage so I decided to gimp the method.
- My thinking is that if the old scripts work, then we probably have better things to be directing efforts to at this stage. If instead, we have issues, I'm seriously considering a refactor, which will include separating out/removing the english word filtering.